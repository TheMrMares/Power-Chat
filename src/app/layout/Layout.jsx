import React from 'react'

export class Layout extends React.Component {
  render() {
    return (
      <div>
        layout
      </div>
    )
  }
}

export default Layout
