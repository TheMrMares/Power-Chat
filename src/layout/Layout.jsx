import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import firebase from '../firebase'
import styles from './Layout.module.scss'

export class Layout extends React.Component {
  static propTypes = {
    getClearSignedUser: PropTypes.func,
  }

  signOut() {
    const { getClearSignedUser } = this.props
    getClearSignedUser()
    firebase.auth().signOut()
  }

  render() {
    return (
      <div className={styles.wrapper}>
        Wrapper
      </div>
    )
  }
}

const mapDispatch = ({
  auth: { getClearSignedUser, getUpdateSignedUser },
}) => ({
  getClearSignedUser,
  getUpdateSignedUser,
})


export default connect(null, mapDispatch)(Layout)
