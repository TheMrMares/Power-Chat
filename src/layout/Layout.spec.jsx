import React from 'react'
import { shallow } from 'enzyme'
import { Layout } from './Layout'

const initPage = overrides => {
  const wrapper = shallow(
    <Layout
      {...overrides}
    />
  )
  return { wrapper }
}

describe('Layout', () => {
  it('should render as expected', () => {
    const { wrapper } = initPage()
    expect(wrapper).toMatchSnapshot()
  })
})
