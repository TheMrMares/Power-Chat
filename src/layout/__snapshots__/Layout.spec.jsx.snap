// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render as expected 1`] = `
<div>
  Wrapper
</div>
`;
